darts/models/theta.py

```
@@ -115,14 +123,228 @@ def predict(self, n: int) -> 'TimeSeries':
+        # - if theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        # - if theta = 0, then the theta method restricts to a simple `trend_mode` regression.
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        self.mean = ts.mean().mean()
+        if self.mean == 0:
+            self.mean = 1.
+        new_ts = ts / self.mean
+
+        if (new_ts <= 0).values.any():
+            self.model_mode = Model.ADDITIVE
+            self.trend_mode = Trend.LINEAR
+            self.season_mode = Season.ADDITIVE
+            logger.warn("Time series has negative values. Fallback to additive and linear model")
+
+        # Check for statistical significance of user-defined season period
+        # or infers season_period from the TimeSeries itself.
+        if self.season_mode is Season.NONE:
+            self.season_period = 1
+        if self.season_period is None:
+            max_lag = len(ts) // 2
+            self.is_seasonal, self.season_period = check_seasonality(ts, self.season_period, max_lag=max_lag)
+            logger.info('Theta model inferred seasonality of training series: {}'.format(self.season_period))
+        else:
+            # force the user-defined seasonality to be considered as a true seasonal period.
+            self.is_seasonal = self.season_period > 1
+
+        # Store and remove seasonality effect if there is any.
+        if self.is_seasonal:
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period,
+                                                                model=self.season_mode.value)
+            new_ts = remove_seasonality(new_ts, self.season_period, model=self.season_mode.value)
+
+        # Drift part of the decomposition
+        if self.trend_mode is Trend.LINEAR:
+            linreg = new_ts.univariate_values()
```

You could start extracting the univariate values earlier in the method, in order to do it only once, and fail early if it's not univariate.